exporter/terra/terra_export_job.py

```diff
@@ -0,0 +1,93 @@
+from dataclasses import dataclass
+from typing import List, Dict, Set
+from ingest.api.ingestapi import IngestApi
+import requests
+
+from enum import Enum
+
+
+@dataclass
+class TerraExportError:
+    message: str
+
+    def to_dict(self) -> Dict:
+        return {
+            "message": self.message,
+            "errorCode": -1,
+            "details": {}
+        }
+
+
+@dataclass
+class TerraExportEntity:
+    assay_process_id: str
+    errors: List[TerraExportError]
+
+    def to_dict(self) -> Dict:
+        """
+        converts to a JSON as represented in ingest-core API
+        """
+        return {
+            "status": ExportJobState.EXPORTED,
+            "context": {
+                "assayProcessId": self.assay_process_id
+            },
+            "errors": [e.to_dict() for e in self.errors]
+        }
+
+
+class ExportJobState(Enum):
+    EXPORTING = "EXPORTING"
+    EXPORTED = "EXPORTED"
+    DEPRECATED = "DEPRECATED"
+    FAILED = "FAILED"
+
+
+@dataclass
+class TerraExportJob:
+    job_id: str
+    num_expected_assays: int
+    export_state: ExportJobState
+
+    @staticmethod
+    def from_dict(data: Dict) -> 'TerraExportJob':
+        job_id = str(data["_link"]["self"]["href"]).split("/")[0]
+        num_expected_assays = int(data["context"]["expectedAssays"])
+        return TerraExportJob(job_id, num_expected_assays, ExportJobState(data["state"]))
+
+
+class TerraExportJobService:
+    def __init__(self, ingest_client: IngestApi):
+        self.ingest_client = ingest_client
+
+    def complete_assay(self, job_id: str, assay_process_id: str):
+        assay_export_entity = TerraExportEntity(assay_process_id, [])
+        create_export_entity_url = self.get_export_entities_url(job_id)
+        requests.post(create_export_entity_url, assay_export_entity.to_dict(), json=True).raise_for_status()
+        self._maybe_complete_job(job_id)
+
+    def _maybe_complete_job(self, job_id):
+        export_job = self.get_job(job_id)
+        if export_job.num_expected_assays == self.get_num_entities_for_job(job_id):
```

Comment:
@aaclan-ebi I think this might be parallel-safe, assuming the number of entities returned by GET /exportJobs/{id}/entities is always consistent with what's in the DB

Reply:
i think export entities should also be created if there are errors. 

This should also check the status of each entities.

Reply:
Good point. Should instead do a query for all export entities that are marked complete/exported